### PR TITLE
Book list request

### PR DESCRIPTION
**Pull request by joacomiguens**
## Summary

- Create Token Request Interceptor.

- Create book services to get books from API.

## Screenshot

books-list

## Trello Card

You can find it [here](https://trello.com/c/WyGKEjZx/12-book-list-auth-header)

### CODE REVIEW

File: wolox-books/src/app/screens/auth/screens/book-list/book-list.component.ts [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
+  constructor(private booksService: BooksService) {}
 
   ngOnInit(): void {
+    this.bookSubscription = this.booksService.getBooks().subscribe((response) => {
+      this.books = response.page;
+    });
   }
 
+  ngOnDestroy(): void {
+    this.bookSubscription.unsubscribe();
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/screens/auth/screens/book-list/book-list.component.ts]: For an http client request don't need to unsubscribe

File: wolox-books/src/app/services/token-interceptor.service.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { IRequestCredentials } from '../models/user.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TokenInterceptorService implements HttpInterceptor {
+  constructor(private injector: Injector) {}
+
+  intercept(req, next) {
+    const userService = this.injector.get(UserService);
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/services/token-interceptor.service.ts:13]: You can inject in the constructor

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

**Review by ebedoyaalzate — COMMENTED** [review]
your image height for books list doesn't have the same height, fix this

File: wolox-books/src/app/screens/auth/screens/book-list/book-list.component.ts (hunk 1 above)
Comment (joacomiguens) [reply wolox-books/src/app/screens/auth/screens/book-list/book-list.component.ts]: Done 😄

File: wolox-books/src/app/screens/auth/screens/book-list/book-list.component.ts [hunk 3]
```diff
@@ -1,15 +1,26 @@
...
 })
 export class BookListComponent implements OnInit {
+  bookSubscription: Subscription;
+  books: Array<IBook>;
 
-  constructor() { }
+  constructor(private booksService: BooksService) {}
 
   ngOnInit(): void {
+    this.bookSubscription = this.booksService.getBooks().subscribe((response) => {
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/screens/auth/screens/book-list/book-list.component.ts:18]: You don`t need bookSubscription for the moment

**Review by ebedoyaalzate — APPROVED** [review]

**Review by mcallegari10 — APPROVED** [review]

**Pull request merged**